src/ray/raylet/scheduling_queue.cc

```diff
@@ -195,27 +195,53 @@ void SchedulingQueue::FilterState(std::unordered_set<TaskID> &task_ids,
   }
 }
 
-std::vector<Task> SchedulingQueue::RemoveTasks(std::unordered_set<TaskID> &task_ids) {
+std::pair<std::vector<Task>, std::vector<TaskState>> SchedulingQueue::RemoveTasks(
+    std::unordered_set<TaskID> &task_ids) {
   // List of removed tasks to be returned.
   std::vector<Task> removed_tasks;
+  std::vector<TaskState> removed_task_states;
+
+  size_t num_queued_tasks = 0;
 
   // Try to find the tasks to remove from the queues.
+
   RemoveTasksFromQueue(methods_waiting_for_actor_creation_, task_ids, removed_tasks);
+  for (; num_queued_tasks < removed_tasks.size(); ++num_queued_tasks) {
```

Comment:
this looks like it should better be done inside RemoveTasksFromQueue

Reply:
Agreed. In general I feel like we should refactor this class to have a map from `TaskState` to `TaskQueue` instead of having individual members for each `TaskQueue` that have no association to the `TaskState` they represent, but we can worry about that later...